### DISCUSSION TITLE
Supporting for Microsoft Dynamics or custom CRM integratioin

### DISCUSSION THREAD

**Discussion by wei2go** [solspace/craft-freeform#1252] [Feature Requests]
Hi Solspace team,

Just wondering if supporting Microsoft Dynamics CRM integration is in your roadmap?

Also, if we want to build a custom CRM integration which integrates with Freeforms (e.g. allow CP users to choose tables and field mappings, etc), what would be your recommendation to start with?

**Comment by gustavs-gutmanis** [comment]
Hi @wei2go 

It isn't on our roadmap yet, as we haven't had enough people show interest in having it.

You are, however, able to build your own integration using the existing tools in Freeform. Here is a guide on creating an [email marketing integration](https://docs.solspace.com/craft/freeform/v5/developer/email-marketing/) which is very similar to making CRM integrations. You could also reference any of our existing CRM integrations to get ideas ([Salesforce](https://github.com/solspace/craft-freeform/tree/v5/packages/plugin/src/Integrations/CRM/Salesforce) for example).

We also do custom development for cases such as yours, but for that, please reach out to us using this form https://docs.solspace.com/support/.

**Reply by wei2go** [reply]
Hi @gustavs-gutmanis ,

Thank you for your reply.

The information for the custom integration is very useful.

Regarding Dynamics, it's quite popular for non-for-profits organisations in Australia, but understand that it may not be as popular in the bigger scale. Hopefully in the future this can be added to your radar.

**Comment by kjmartens** [comment]
The next version of Freeform will include a beta integration for MS Dynamics 365 🙂

**Comment by kjmartens** [comment]
This has now been added in Freeform 5.16.0 🙂
